.github/workflows/publish.yml

```diff
@@ -1,21 +1,30 @@
 name: Publish crates
 
 on:
+  # We only want to perform dry runs when this workflow runs on PRs.
+  pull_request:
+    branches: [master]
```

Comment:
As @TomAFrench called out, this will need to become `pull_request_target` before merge